docs/project/os-onboarding.md

```diff
@@ -0,0 +1,105 @@
+
+We use both containers and VMs for building, depending on the OS. If we test in a container, we likely build in a container. Same for VMs.
+
+Our primary concern is ensuring that we are using [supported operating systems and tools for our build](https://github.com/dotnet/runtime/tree/main/docs/workflow/requirements).
+
+Our Linux build containers are based on Azure Linux. We [typically need to update them](https://github.com/dotnet/runtime/issues/112191) with a new version of Azure Linux once per release. We do not update the toolset, however. That's fixed, per release.
+
+For Apple, we likely need to make an adjustment at each macOS or iOS release to account for an XCode version no longer being supported.
+
+## Prereqs containers
+
+New images need to be created for each new OS version in the [dotnet/dotnet-buildtools-prereqs-docker](https://github.com/dotnet/dotnet-buildtools-prereqs-docker) repo, for testing. We also need to create and update images for Linux and Wasm build environments.
+
+The repo is self-service and largely self-explanatory. One typically creates a new image using the pattern demonstrated by the previous version. Look at commits and [blame](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blame/776324ff16d38e22fd9f06c9842ec338a4b98489/src/alpine/3.20/helix/Dockerfile) to find people who are best suited to help.
+
+Installing/building the Helix client can be quite involved, particularly for Arm platforms. Don't struggle with that. Just ask for help.
+
+Container images are referenced in our pipeline files:
+
+- [eng/pipelines/coreclr/templates/helix-queues-setup.yml](https://github.com/dotnet/runtime/blob/main/eng/pipelines/coreclr/templates/helix-queues-setup.yml)
+- [eng/pipelines/libraries/helix.yml](https://github.com/dotnet/runtime/blob/main/eng/pipelines/libraries/helix.yml)
+- [eng/pipelines/common/templates/pipeline-with-resources.yml](https://github.com/dotnet/runtime/blob/main/eng/pipelines/common/templates/pipeline-with-resources.yml)
+
+Notes:
+
+- The first two links are for testing and the last for building.
+- The links are for the `main` branch. Release branches should have the same layout.
+
+Example PRs:
+
+- [dotnet/runtime #111768](https://github.com/dotnet/runtime/pull/111768)
+- [dotnet/runtime #111504](https://github.com/dotnet/runtime/pull/111504)
+- [dotnet/runtime #110492](https://github.com/dotnet/runtime/pull/110492)
+- [dotnet/dotnet-buildtools-prereqs-docker #1282](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1282)
+- [dotnet/dotnet-buildtools-prereqs-docker #1314](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1314)
+
+### VMs
+
+VMs and raw metal environments are used for Linux, macOS, and Windows. They need to be [requested from dnceng](https://github.com/dotnet/dnceng/issues/4307). The turnaround can be long so put in your request before you need it.

```

Comment:
Android is on the raw metal plan as well: https://github.com/dotnet/runtime/blob/497236a2359f9de4b70b1dbd00e55449823e6f63/eng/pipelines/libraries/helix-queues-setup.yml#L94-L97